virttest/utils_test/libvirt.py

```diff
@@ -595,21 +596,36 @@ def setup_or_cleanup_gluster(is_setup, vol_name, brick_path="", pool_name="",
     if not brick_path:
         tmpdir = data_dir.get_tmp_dir()
         brick_path = os.path.join(tmpdir, pool_name)
+
+    # Check gluster server apply or not
+    ip_addr = kwargs.get("gluster_server_ip", "")
+    session = None
+    if ip_addr != "":
+        remote_user = kwargs.get("gluster_server_user")
+        remote_pwd = kwargs.get("gluster_server_pwd")
+        session = remote.remote_login("ssh", ip_addr, "22",
+                                      remote_user, remote_pwd, "#")
     if is_setup:
-        ip_addr = get_host_ipv4_addr()
-        gluster.add_rpc_insecure(file_path)
-        gluster.glusterd_start()
-        logging.debug("finish start gluster")
-        gluster.gluster_vol_create(vol_name, ip_addr, brick_path, force=True)
-        gluster.gluster_allow_insecure(vol_name)
-        gluster.gluster_nfs_disable(vol_name)
+        if ip_addr == "":
+            ip_addr = get_host_ipv4_addr()
+            gluster.add_rpc_insecure(file_path)
+            gluster.glusterd_start()
+            logging.debug("finish start gluster")
+
+        gluster.gluster_vol_create(vol_name, ip_addr, brick_path, True, session)
+        gluster.gluster_allow_insecure(vol_name, session)
+        gluster.gluster_nfs_disable(vol_name, session)
         logging.debug("The contents of %s: \n%s", file_path, open(file_path).read())
```

Comment:
with 'open(file_path)' will open a local file. Will  "/etc/glusterfs/glusterd.vol" be in local machine during test? Our test on p9 has failed as "No such file or directory: '/etc/glusterfs/glusterd.vol'"

Reply:
fixed, thanks